lib/Option.cpp

```diff
@@ -168,6 +168,18 @@ llvm::cl::opt<clspv::Option::SourceLanguage> cl_std(
 static llvm::cl::opt<bool> images("images", llvm::cl::init(true),
                                   llvm::cl::desc("Enable support for images"));
 
+static llvm::cl::opt<bool>
+    scalar_block_layout("scalar-block-layout", llvm::cl::init(false),
+                        llvm::cl::desc("Assume VK_EXT_scalar_block_layout"));
+
+static llvm::cl::opt<bool> work_dim(
+    "work-dim", llvm::cl::init(false),
+    llvm::cl::desc("Enable support for get_work_dim() built-in function"));
+
+static llvm::cl::opt<bool>
+    global_offset("global-offset", llvm::cl::init(false),
```

Comment:
I assume this is intended to default on at some point in the future (e.g. when support is landed in clvk)?

Reply:
Maybe but that would mean all the applications that use clspv have to support and pass push constants.

Reply:
Only if those builtins are used. I don't think that's a problem since push constants are a core feature.

Reply:
Currently the descriptor map entries are always present but we could only emit them when the built-in is used for `get_work_dim`. For `get_global_offset` however, the entry would always be present if `get_global_id` is used. I'm not worried about optionality, just forcing applications to support something they may not need. Having said that, using push constants for POD arguments might be a good justification for requiring applications to pass push constants.